mlflow/system_metrics/metrics/base_metrics_monitor.py

```diff
@@ -0,0 +1,32 @@
+"""Base class of system metrics monitor."""
+from collections import defaultdict
+
+
+class BaseMetricsMonitor:
+    """Base class of system metrics monitor.
+
+    Args:
+        name: string, name of the monitor.
+    """
+
+    def __init__(self, name):
+        self.name = name
+
+        self._metrics = defaultdict(list)
+
+    def collect_metrics(self):
+        raise NotImplementedError
+
+    def aggregate_metrics(self):
+        metrics = {}
+        for name, values in self._metrics.items():
+            if len(values) > 0:
+                metrics[name] = sum(values) / len(values)
+        return metrics
```

Comment:
```suggestion
        from statistics import mean 

        return {k: mean(vs) for k, vs in self._metrics.items() if vs}
```

Reply:
`mean([])` returns a failure, we still need the empty check.